design/external-references.md

```diff
@@ -0,0 +1,394 @@
+# RFC: External Resource References
+
+## Background
+
+PR [#1436](https://github.com/awslabs/aws-cdk/pull/1436) had recently introduced implicit
+references of resource attributes across stacks within the same CDK app:
+
+
+```ts
+const user = new iam.User(stackWithUser, 'User');
+const group = new iam.Group(stackWithGroup, 'Group');
+
+group.addUser(user);
+```
+
+Will synthesize the following two templates:
+
+```yaml
+Outputs:
+  ExportsOutputRefGroupC77FDACD8CF7DD5B:
+    Value:
+      Ref: GroupC77FDACD
+    Export:
+      Name: ExportsOutputRefGroupC77FDACD8CF7DD5B
+Resources:
+  GroupC77FDACD:
+    Type: AWS::IAM::Group
+```
+
+and:
+
+```yaml
+Resources:
+  User00B015A1:
+    Type: AWS::IAM::User
+    Properties:
+      Groups:
+      - Fn::ImportValue: ExportsOutputRefGroupC77FDACD8CF7DD5B
+```
+
+You can see that the reference automatically resulted in an `Output+Export` and
+an `Fn::ImportValue` to be used in order to share this information across the
+stacks within the same application.
+
+This mechanism practically replaces our "import/export" pattern which looks like this (same use case and same synthesized output):
+
+```ts
+const user = new iam.User(stackWithUser, 'User');
+const group = new iam.Group(stackWithGroup, 'Group');
+
+// creates an `Output` in "stackWithUser" with a unique export name
+// returns an `Fn::ImportValue` that can be used to import this user
+// in any stack in the same account/region
+const importValues = user.export();
+
+// instantiates an `IUser` whose attributes are `Fn::ImportValue` tokens
+// in "stackWithGroup", and it can now be used "naturally" in that stack
+const importedUser = iam.User.import(stackWithGroup, 'ImportedUser', importValues);
+group.addUser(importedUser);
+```
+
+Obvsiouly the world is a better place now. Thanks [@rix0rrr](https://github.com/users/rix0rrr).
+
+## Problem Statement
+
+So, "what seems to be the problem?", you ask. Well, we want more! The current model only supports references that occur between stacks within the same CDK app. What about the use case where I want to reference a resource that was created in a totally different app?
+
+A common example is for a common infrastructure team to maintain a CDK app which defines networking resources such as VPCs, security groups, LBs, etc and application teams need to reference these resources in their own CDK apps.
+
+__So the problem we are trying to solve is how to enable users to reference resources (and potentially higher level constructs) across CDK apps.__
+
+## Scope
+
+As an initial step, we'll address the use case of referencing resources across within __the same AWS environment__ (account + region). This limitation is based on the limitation AWS CloudFormation has for exports/imports.
+
+We differentiate two types of references:
+
+1. Reference a resource created through another CDK app.
+2. Reference a resource created by some other means.
```

Comment:
"CloudFormation-not-CDK" could possibly "manually" create the required Outputs/exports and refer to them from the CDK app. This would mean the naming protocol needs to be something that can be manually derived, ideally.